packages/vhd-lib/Vhd/VhdAbstract.js

```diff
@@ -301,10 +301,16 @@ exports.VhdAbstract = class VhdAbstract {
 
     assert.strictEqual(offset % SECTOR_SIZE, 0)
     const self = this
+    let yielded = 0
+    function* trackAndYield(buffer) {
+      yielded += buffer.length
+      assert.ok(yielded <= fileSize, `Max stream length is ${fileSize}, try to send ${yielded}`)
+      yield buffer
+    }
```

Comment:
I'm curious/worried of the perf impact of this.

Reply:
But, we don't have the tools to measure or improve that for now.

Let's not worry unnecessarily.

Reply:
We can optimize slightly:

```js
function trackAndYield(buf) {
  // todo
  return buf
}

yield trackAndYield(buf)
```